celery_utils/logged_task.py

```diff
@@ -0,0 +1,34 @@
+"""
+Improved logging for celery tasks.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+from celery import Task
+
+
+log = logging.getLogger(__name__)
+
+
+# pylint: disable=abstract-method
+class LoggedTask(Task):
+    """
+    Task base class that emits a log statement when it gets submitted.
+    """
+
+    abstract = True
+
+    def apply_async(self, args, kwargs, **options):
+        """
+        Emit a log statement when the task is submitted.
+        """
+        result = super(LoggedTask, self).apply_async(args, kwargs, **options)
+        log.info('Task {}[{}] submitted with arguments {}, {}'.format(
+            self.name,
+            result.id,
+            args,
+            kwargs
+        ))
```

Comment:
I just added a commit that logs the exception on retry.  However, I was unable to make celery actually call the on_retry handler, because is_eager shortcuts that codepath.